### PR TITLE
[ci] check PowerShell scripts with PSScriptAnalyzer (part 2)

### PR DESCRIPTION
**Pull request by StrikerRUS**
Linking full original PR: #6700.
Continuation of #6704.

This PR fixes only all `PSUseConsistentIndentation` warnings, so it can be skimmed fast without attention.

### CODE REVIEW

File: .ci/test.sh [hunk 1]
```diff
@@ -100,7 +100,7 @@ fi
 if [[ $TASK == "lint" ]]; then
     pwsh -command "Install-Module -Name PSScriptAnalyzer -Scope CurrentUser -SkipPublisherCheck"
     echo "Linting PowerShell code"
-    pwsh -file "./.ci/lint-powershell.ps1" || exit 0
+    pwsh -file "./.ci/lint-powershell.ps1" || :
```
Comment (StrikerRUS) [inline .ci/test.sh:103]: Just noticed that after [moving powershell linter upper all other ones](https://github.com/microsoft/LightGBM/pull/6704#discussion_r1821440350), `|| exit 0` causes terminating the whole script earlier than running all linters.

Ref. for `:`
- https://stackoverflow.com/a/33427572.

Comment (jameslamb) [reply .ci/test.sh:103]: ah right! Sorry 😅 

`|| true` would work too. I'm indifferent, since this is temporary anyway

Comment (jameslamb) [reply .ci/test.sh:103]: I checked the `lint` job logs ([build link](https://github.com/microsoft/LightGBM/actions/runs/11620094860/job/32361143252?pr=6709)) and can confirm all the other linting is running, so looks like this works. Thanks very much for the Stack Overflow link, I hadn't seen `|| :` before.

**Review by jameslamb — APPROVED** [review]
Thanks very much for doing the work of breaking these into smaller PRs! I'm glad to see this standardization of indentation in these scripts too 😁

File: .ci/test.sh (hunk 1 above)
Comment (jameslamb) [reply .ci/test.sh:103]: Oh and I hadn't yet clicked that S/O link before suggesting `|| true`. I see it says there:

> *The above suggestions to use 'true' will also work, but are inefficient as 'true' is an external program.*

That's good to know! I have been using `|| true` as a habit for a long tim.

Comment (StrikerRUS) [reply .ci/test.sh:103]: > I have been using `|| true` as a habit for a long tim.

Me too. And hah, seems it's OK - just noticed the following comment under the S/O answer:
> True is only an external program if you run `/bin/true`, as `true` on its own is a shell built-in of bash, just like echo and test, which also exists as external programs but are also built-ins in bash and if you don't give a full path, the built-ins will be used. 

https://stackoverflow.com/questions/17830326/ignoring-specific-errors-in-a-shell-script/33427572#comment69729658_33427572

Comment (jameslamb) [reply .ci/test.sh:103]: oh wow!

**Pull request merged**